azurerm/resource_arm_batch_pool.go

```diff
@@ -0,0 +1,476 @@
+		targetLowPriorityNodes := int32(d.Get("target_low_priority_nodes").(int))
+		resizeTimeout := d.Get("resize_timeout").(string)
+
+		parameters.PoolProperties.ScaleSettings = &batch.ScaleSettings{
+			FixedScale: &batch.FixedScaleSettings{
+				ResizeTimeout:          &resizeTimeout,
+				TargetDedicatedNodes:   &targetDedicatedNodes,
+				TargetLowPriorityNodes: &targetLowPriorityNodes,
+			},
+		}
+	} else {
+		return fmt.Errorf("Error: scale mode should be either AutoScale of FixedScale")
+	}
+
+	if _, err = client.Update(ctx, resourceGroup, accountName, poolName, parameters, ""); err != nil {
+		return fmt.Errorf("Error creating Batch pool %q (Resource Group %q): %+v", poolName, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, accountName, poolName)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Batch pool %q (Resource Group %q): %+v", poolName, resourceGroup, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Batch pool %q (resource group %q) ID", poolName, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmBatchPoolRead(d, meta)
+}
+
+func resourceArmBatchPoolDelete(d *schema.ResourceData, meta interface{}) error {
+	ctx := meta.(*ArmClient).StopContext
+	client := meta.(*ArmClient).batchPoolClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	poolName := id.Path["pools"]
+	accountName := id.Path["batchAccounts"]
+
+	future, err := client.Delete(ctx, resourceGroup, accountName, poolName)
+	if err != nil {
+		return fmt.Errorf("Error deleting Batch pool %q (Resource Group %q): %+v", poolName, resourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error waiting for deletion of Batch pool %q (Resource Group %q): %+v", poolName, resourceGroup, err)
+		}
+	}
+	return nil
+}
+
+func resourceArmBatchPoolRead(d *schema.ResourceData, meta interface{}) error {
+	ctx := meta.(*ArmClient).StopContext
+	client := meta.(*ArmClient).batchPoolClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	poolName := id.Path["pools"]
+	accountName := id.Path["batchAccounts"]
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, poolName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Error: Batch pool %q in account %q (Resource Group %q) was not found", poolName, accountName, resourceGroup)
+		}
+		return fmt.Errorf("Error making Read request on AzureRM Batch pool %q: %+v", poolName, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	d.Set("name", poolName)
+	d.Set("account_name", accountName)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("vm_size", resp.VMSize)
+
+	if resp.ScaleSettings != nil {
+		if resp.ScaleSettings.AutoScale != nil {
```

Comment:
As above, i think these should be in two separate blocks

Reply:
Done.